tests/hla/base_test.py

```diff
@@ -0,0 +1,405 @@
+
+            fd.write("1 match\n")
+            fd.write("distance = %f flux difference = %f\n" % (distances[0][0], distances[0][1]))
+
+            for j in range(2, 4):
+                ylo = int(distances[0][j][0]) - (1+self.buff)
+                yhi = int(distances[0][j][0]) + (2+self.buff)
+                xlo = int(distances[0][j][1]) - (1+self.buff)
+                xhi = int(distances[0][j][1]) + (2+self.buff)
+                subimage = images[j][ylo:yhi,xlo:xhi]
+                fd.write("\n%s image centroid = (%f,%f) image flux = %f\n" %
+                         (im_type[j], distances[0][j][0], distances[0][j][1], distances[0][j][2]))
+                fd.write(str(subimage) + "\n")
+
+        else:
+            fd.write("%d matches\n" % len(distances))
+
+            for k in range(0,3):
+                i = indexes[k]
+                diff.append(distances[i][0])
+                fd.write("\n%s distance = %f flux difference = %f\n" % (stats[k], distances[i][0], distances[i][1]))
+
+                for j in range(2, 4):
+                    ylo = int(distances[i][j][0]) - (1+self.buff)
+                    yhi = int(distances[i][j][0]) + (2+self.buff)
+                    xlo = int(distances[i][j][1]) - (1+self.buff)
+                    xhi = int(distances[i][j][1]) + (2+self.buff)
+                    subimage = images[j][ylo:yhi,xlo:xhi]
+                    fd.write("\n%s %s image centroid = (%f,%f) image flux = %f\n" %
+                             (stats[k], im_type[j], distances[i][j][0], distances[i][j][1], distances[i][j][2]))
+                    fd.write(str(subimage) + "\n")
+
+        fd.close()
+        return tuple(diff)
+
+    def make_point_image(self, input_image, point, value):
+        """
+        Create an image with a single point set
+        """
+        output_image = np.zeros(input_image.shape, dtype=input_image.dtype)
+        output_image[point] = value
+        return output_image
+
+    def make_grid_image(self, input_image, spacing, value):
+        """
+        Create an image with points on a grid set
+        """
+        output_image = np.zeros(input_image.shape, dtype=input_image.dtype)
+
+        shape = output_image.shape
+        for y in range(spacing//2, shape[0], spacing):
+            for x in range(spacing//2, shape[1], spacing):
+                output_image[y,x] = value
+
+        return output_image
+
+    def print_wcs(self, title, wcs):
+        """
+        Print the wcs header cards
+        """
+        print("=== %s ===" % title)
+        print(wcs.to_header_string())
+
+
+    def read_image(self, filename):
+        """
+        Read the image from a fits file
+        """
+        hdu = fits.open(filename)
+
+        image = hdu[1].data
+        hdu.close()
+        return image
+
+    def read_wcs(self, filename):
+        """
+        Read the wcs of a fits file
+        """
+        hdu = fits.open(filename)
+
+        wcs = stwcs.wcsutil.HSTWCS(hdu, 1)
+        hdu.close()
```

Comment:
Worse:
```python
with fits.open(filename) as hdu:
    wcs = stwcs.wcsutil.HSTWCS(hdu, 1)
```
Best (option): `HSTWCS` can open files itself:
```python
wcs = stwcs.wcsutil.HSTWCS(filename, 1)
```
Given this, is `read_wcs()` even necessary?

Reply:
Deferred to Git Issue #295.

Reply:
Implemented in #309